frontend/src/metabase/visualizations/components/ColumnSettings.jsx

```diff
@@ -56,6 +56,7 @@ function getWidgets({
 
   return widgets.filter(
     widget =>
+      !widget.hidden &&
```

Comment:
this is the fix: we simply weren't even checking if widgets should be hidden

Reply:
Currently, the logic to [show/hide widgets](https://github.com/metabase/metabase/blob/1efa6c6f000cb4bd38065563176c35ea0027c865/frontend/src/metabase/visualizations/components/ChartSettingsWidget.tsx#L59) is in `ChartSettingsWidget`. Is there a reason that wasn't kicking in?